backend/src/main/java/com/redhat/cloud/notifications/oapi/OApiFilter.java

```diff
@@ -0,0 +1,203 @@
+package com.redhat.cloud.notifications.oapi;
+
+import com.redhat.cloud.notifications.Constants;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.ext.web.client.WebClient;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.WebApplicationException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class OApiFilter {
+
+    public static final String NOTIFICATIONS = "notifications";
+    public static final String INTEGRATIONS = "integrations";
+    public static final String PRIVATE = "private";
+    public static final String INTERNAL = "internal";
+
+    static List<String> whats = List.of(INTEGRATIONS, NOTIFICATIONS, PRIVATE, INTERNAL);
+
+    @Inject
+    Vertx vertx;
+
+    @ConfigProperty(name = "quarkus.http.port", defaultValue = "8085")
+    Integer port;
+
+    private WebClient client;
+
+    @PostConstruct
+    void initialize() {
+        this.client = WebClient.create(vertx,
+                new WebClientOptions().setDefaultHost("localhost").setDefaultPort(port));
+    }
+
+    public Uni<String> serveOpenApi(String what) {
+        if (!whats.contains(what)) {
+            throw new WebApplicationException("No openapi file for [" + what + "] found.", 404);
+        }
+
+        return client.get("/openapi.json")
+                .send()
+                .onItem()
+                .transform(response ->
+                        filterJson(response.bodyAsJsonObject(), what)
+                )
+                .onItem().transform(JsonObject::encode);
+    }
+
+    private JsonObject filterJson(JsonObject oapiModelJson, String what) {
+
+        JsonObject root = new JsonObject();
+
+        oapiModelJson.stream().forEach(entry -> {
+            String key = entry.getKey();
+            switch (key) {
+                case "components":
+                case "openapi":
+                    // We just copy all of them even if they may only apply to one
+                    root.put(key, entry.getValue());
+                    break;
+                case "tags":
+                    JsonArray tags = (JsonArray) entry.getValue();
+                    JsonArray filteredTags = new JsonArray(tags
+                            .stream()
+                            .filter(o -> !((JsonObject) o).getString("name").equals(PRIVATE))
```

Comment:
could getString() lead to an NPE here?

Reply:
tags should always have `name` field. I think is OK to fail if that's not true.

https://swagger.io/specification/#tag-object